src/web/rustdoc.rs

```diff
@@ -493,6 +493,11 @@ fn path_for_version(
     } else {
         ""
     };
+    let is_source_view = if platform.is_empty() {
+        req_path.get(3).copied() == Some("src")
```

Comment:
without knowing all the connections too well, quite some digging is necessary to know which two cases we are handling here. 

Could we add a comment here?

Reply:
Sure, I left a couple comments - let me know if they make sense.